v3.0_roadmap.adoc

```diff
@@ -0,0 +1,243 @@
+
+|ElasticSearch
+|supported
+|
+
+|HBase
+|unsupported
+|
+
+|JCR
+|unsupported
+|
+
+|Lucene
+|unsupported
+|
+
+|Caching
+|unsupported
+|
+
+|Maildir
+|unsupported
+|Known issue on MailboxPath handling
+
+|===
+
+What being "supported implies for a backend ?
+
+ - Being MPT tested
+ - Deployement tests part of unit tests (ex : cassandra-guice)
+ - Having full support in server/data/data-@backend@
+ - Having its mappers passing generic unit tests for mappers
+ - It should at least support USER_FLAGS and NAMESPACE features on MPT tests
+ - Be able to handle a constant gatling workload for at least 4 hours with constant latency
+ - Be tested with thunderbird (we should write scenari for that)
+
+
+.Feature matrix : mailbox capabilities
+|===
+|Feature name |Status | Comment
+
+|ACL
+|unsupported
+|
+
+|Shared mailbox
+|unsupported
+|
+
+|Quota
+|experimental
+|
+
+|User flags
+|supported
+|
+
+|Distributed events
+|Unsupported
+|
+
+|===
+
+=== Short note about what needs to be DONE
+
+**General** :
+
+ - Define Thunderbird scenari
+ - Write Gatling tests for IMAP
+
+**Cassandra** :
+
+ - We need a Cassandra implementation for MailRepository
+
+**Memory** :
+
+ - We need a Memory SieveRepository
+ - We need a Memory implementation for MailRepository
+
+**JPA** :
+
+ - We need JAMES-1693-2 to be merged.
+ - We need JPA implementation for SieveRepository and MailRepository.
+ - We need JPA to valid MessageMapper and MailboxMapper unit tests.
+
+Bonus : have quota support with JPA...
+
+== Server
+
+.Injection framework
+|===
+|Name |Status
+
+|Spring
+|Supported
+
+|Guice cassandra
+|experimental
+
+|Guice memory
+|experimental
+
+|Cluster deployement
+|unsupported
+|===
+
+We need tests to ensure James starts when Spring is used.
+
+.Components
+|===
+|Name |Status |Comment
+
+|ActiveMQ mail queue
+|supported
+|
+
+|Camel mail processing
+|supported
+|
+
+|Sieve
+|experimental
+|Problem : it is compulsory today to rely on SIEVE scripts upon message reception
+
+|Mailets
+|experimental / supported
+|If need be, I can add the details on mailet
+|===
+
+=== What to do ?
+
+Integration test on Spring. We need to ensure it starts in JPA flavour with some supported components.
+
+Write a LocalDelivery mailet variant that do not rely on SieveMailet, to make SIEVE use truly optional.
+
+== Protocols
+
+.Protocols
+|===
+|Name |Status |Comment
+
+|IMAP
+|supported
+|See details on IMAP extensions below
+
+|SMTP
+|supported
+|Lack some integration tests, I guess
+
+|POP3
+|Unsupported
+|
+
+|ManageSIEVE
+|Unsupported
+|
+
+|JMAP
+|Experimental
+|
+
+|LMTP
+|supported
+|
+
+
+|FetchMail
+|Unsupported
+|
+|===
+
+
+.IMAP extensions
+|===
+|Header 1 |Header 2 |Header 3
+
+|RFC-2087 Quota
+|experimental
+|
+
+|RFC-4314 ACLs
+|Unsupported
+|
+
+|RFC-5464 Metadatas
+|Unsupported
+|
+
+|RFC-3051 IMAP4 rev1
+|supported
+|Following https://james.apache.org/protocols/imap4.html
+
+|RFC-2342 Namespace
+|supported
```

Comment:
In fact having this without proper ACL support is dangerous IMO. 
Even if implemented, We should not activate it until ACL are ready.

I will add a TODO on this.